posts/2022-03-11-dev-mode-with-eclipse.adoc

```diff
@@ -0,0 +1,101 @@
+++++
+
+== Working with dev mode in the Eclipse IDE with Liberty Developer Tools installed
+
+If you have Liberty Developer Tools installed in your Eclipse workspace, the server configuration editor views can be helpful as you modify your `server.xml` and other configuration files.
+
+image::/img/blog/liberty-dev-tools-server-config-view.png[Liberty Developer Tools server configuration view for the Eclipse IDE,width=70%,align="center"]
+
+Dev mode handles a lot of the heavy lifting for developers when using the Liberty runtime. With dev mode, you no longer need to manage your Liberty instances through the "Servers" view of Liberty Developer Tools.
+Disable Liberty Developer Tools' integration for Maven and Gradle projects:
+
+* For Maven projects: **Window -> Preferences -> Maven -> Liberty Integration** drop-down to "No".
+
+* For Gradle projects: **Window -> Preferences -> Gradle -> Liberty Integration** drop-down to "No".
+
+image::/img/blog/liberty-dev-tools-disable-servers-view.png[Liberty Developer Tools disabling the Servers view for the Eclipse IDE,width=70%,align="center"]
+
+== Setting up the project
+
+For Maven projects, specify the Liberty Maven Plugin in your project's `pom.xml` file:
+[source,xml]
+----
+    <plugin>
+        <groupId>io.openliberty.tools</groupId>
+        <artifactId>liberty-maven-plugin</artifactId>
+        <version>3.5.1</version>
+    </plugin>
+----
+
+For Gradle projects, specify the Liberty Gradle Plugin in your project's `build.gradle` file:
+[source,groovy]
+----
+buildscript {
+    repositories {
+        mavenCentral()
+    }
+    dependencies {
+        classpath 'io.openliberty.tools:liberty-gradle-plugin:3.3'
+    }
+}
+----
+
+== Interacting with dev mode
+=== Starting dev mode
+
+Start dev mode through the Eclipse IDE terminal (Navigate to your project in the package explorer -> Right-click -> "Show in Terminal") or through your operating system's terminal with the following commands:
+
+* Maven: `mvn liberty:dev`
+* Gradle: `gradle libertyDev`
+
+image::/img/blog/dev-mode-eclipse-ide-terminal.png[Starting Liberty dev mode through the Eclipse IDE terminal,width=70%,align="center"]
+
+=== Running tests
+
+You can run tests on demand by pressing Enter in the terminal running dev mode.
+
+=== Attaching a debugger
+
+You can attach a debugger to the running server to step through your code at any time. In Eclipse you can create a debug configuration by selecting "Run" -> "Debug Configurations..." -> "Remote Java application". The default port for debugging is `7777`.
+
+image::/img/blog/dev-mode-eclipse-ide-debug-config.png[Liberty debug configuration for the Eclipse IDE,width=70%,align="center"]
+
+=== Stopping dev mode
+
+You can quit dev mode at any time by pressing CTRL+C in the terminal, or type 'q' and press Enter.
+
```

Comment:
can you do this from Eclipse?

Reply:
This may be the Eclipse IDE terminal or an operating system terminal referenced in the "Starting dev mode" section above.